pkg/controller/datavolume-controller.go

```diff
@@ -1609,29 +1610,58 @@ func (r *DatavolumeReconciler) validateSameVolumeMode(
 }
 
 func (r *DatavolumeReconciler) validateAdvancedCloneSizeCompatible(
+	dataVolume *cdiv1.DataVolume,
 	sourcePvc *corev1.PersistentVolumeClaim,
 	targetStorageSpec *corev1.PersistentVolumeClaimSpec) (bool, error) {
-
 	srcStorageClass := &storagev1.StorageClass{}
 	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: *sourcePvc.Spec.StorageClassName}, srcStorageClass); IgnoreNotFound(err) != nil {
 		return false, err
 	}
 
+	srcRequest, hasSrcRequest := sourcePvc.Spec.Resources.Requests[corev1.ResourceStorage]
```

Comment:
very good question, 
and yes, because this is the actual PVC, not the fields from DataVolume, and the PVC is resized by changing the actual request size on the PVC.Spec.Resources